WordPress/Classes/System/RootViewCoordinator.swift

```diff
@@ -34,33 +45,58 @@ class RootViewCoordinator {
 
     // MARK: Private instance variables
 
-    private(set) var rootViewPresenter: RootViewPresenter
+    private var rootViewPresenter: RootViewPresenter?
     private var currentAppUIType: AppUIType {
         didSet {
             updateJetpackFeaturesRemovalCoordinatorState()
         }
     }
     private var featureFlagStore: RemoteFeatureFlagStore
     private var windowManager: WindowManager?
+    private let wordPressAuthenticator: WordPressAuthenticatorProtocol.Type
 
     // MARK: Initializer
 
     init(featureFlagStore: RemoteFeatureFlagStore,
-         windowManager: WindowManager?) {
+         windowManager: WindowManager?,
+         wordPressAuthenticator: WordPressAuthenticatorProtocol.Type = WordPressAuthenticator.self) {
         self.featureFlagStore = featureFlagStore
         self.windowManager = windowManager
         self.currentAppUIType = Self.appUIType(featureFlagStore: featureFlagStore)
-        switch self.currentAppUIType {
+        self.wordPressAuthenticator = wordPressAuthenticator
+        updateJetpackFeaturesRemovalCoordinatorState()
+    }
+
+    // MARK: - Root Coordination
+
+    func showAppUI(animated: Bool = true, completion: (() -> Void)? = nil) {
+        let rootViewPresenter = createPresenter(currentAppUIType)
+        windowManager?.show(rootViewPresenter.rootViewController, animated: animated, completion: completion)
+        self.rootViewPresenter = rootViewPresenter
+
+        updatePromptsIfNeeded()
+    }
+
+    func showSignInUI(completion: (() -> Void)? = nil) {
+        guard let loginViewController = wordPressAuthenticator.loginUI() else {
+            fatalError("No login UI to show to the user.  There's no way to gracefully handle this error.")
+        }
+
+        windowManager?.show(loginViewController, completion: completion)
+        wordPressAuthenticator.track(.openedLogin)
+        self.rootViewPresenter = nil
```

Comment:
This looks to be the core of the fix here. I think it will deallocate anything `WPTabBarController` and beyond once we fix the other retain cycles we've talked about 👍

Reply:
Yes, it's a core fix. However, if there're retain cycles in other view controllers (and there were plenty), `WPTabBarController` gets deallocated but other ViewControllers and the resources they hold - don't.